app/controllers/administrative-units/administrative-unit/sites/site/edit.js

```diff
@@ -0,0 +1,76 @@
+import Controller from '@ember/controller';
+import { dropTask } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class AdministrativeUnitsAdministrativeUnitSitesSiteEditController extends Controller {
+  @service router;
+  @tracked isPrimarySite =
+    this.site.get('id') === this.administrativeUnit.get('primarySite.id');
+
+  get site() {
+    return this.model.adminUnitSite.site;
+  }
+
+  get administrativeUnit() {
+    return this.model.adminUnitSite.administrativeUnit;
+  }
+
+  @dropTask
+  *save(event) {
+    event.preventDefault();
+
+    let address = yield this.site.address;
+
+    if (address.hasDirtyAttributes) {
+      let fullStreet = `${address.street || ''} ${address.number || ''} ${
+        address.boxNumber || ''
+      }`.trim();
+
+      let muncipalityInformation = `${address.postcode || ''} ${
+        address.municipality || ''
+      }`.trim();
+
+      if (fullStreet && muncipalityInformation) {
+        address.fullAddress = `${fullStreet}, ${muncipalityInformation}`;
+      } else if (fullStreet) {
+        address.fullAddress = fullStreet;
+      } else {
+        address.fullAddress = muncipalityInformation;
+      }
+
+      yield address.save();
+    }
+
+    let contacts = yield this.site.contacts;
+    if (contacts.firstObject.hasDirtyAttributes) {
+      yield contacts.firstObject.save();
+    }
+
+    yield this.site.save();
+
+    if (!this.isPrimarySite) {
```

Comment:
I think this primary site logic isn't 100% solid yet. If the user checks the edit page of a non-primary site, and saves the site, it will remove the actual (other) primary site relationship, right? So after that the admin unit won't have a primary site anymore? I might be missing something though.

Reply:
I think this should check if the site was the primary site _before_ the user entered the page and if it was, only then remove the primary site relationship?

Reply:
Yes, I forgot about that use case.

Reply:
Just a summary to see if I understood it correctly:

Keeping the current data model in mind (primary-site isn't part of sites) the following scenario's are possible:

1. The user edits the site that is also the primary site at the moment:
    1. It's still marked as primary site on save -> nothing needs to move around
    2. It's not marked as primary anymore
        - primary site relationship needs to be cleared
        - the site record needs to be added to the sites relationship
        - (This scenario is problematic since we use the primary site in multiple places :smile:.)
2. The user edits a site that isn't the primary site at the moment
    1. It's now marked as the primary site on save 
        - the record that's currently linked as primary site needs to be added to the sites relationship
        - the record that's being edited needs to be set as the primary site record and removed from the sites relationship
     2. It's not marked as primary on save -> nothings needs to move around

Reply:
If we show the info we have, I think it won't be a problem. If the user doesn't select a primary site, it's not a bug.

Reply:
To confirm, I need to change the first if only to change it if before it was a primary site.